policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/stream/config/AkkaRunnableGraph.java

```diff
@@ -70,68 +72,64 @@ KafkaProducerService kafkaProducerService(
 
     @Bean
     Function1<Throwable, Directive> supervisor() {
-        return ex -> Supervision.resumingDecider().apply(ex);
+        return (Throwable ex) -> {
+            LOGGER.error("Fatal error during stream processing, element will be dropped: ", ex);
+            return Supervision.resumingDecider().apply(ex);
+        };
     }
 
     @Bean
     RunnableGraph<DrainingControl<Done>> runner(
             final Source<CommittableMessage<String, PolicyRequest>, Control> source,
-            final Sink<Envelope<String, PolicyResponse, Committable>, CompletionStage<Done>> sink,
+            final Sink<Envelope<String, byte[], Committable>, CompletionStage<Done>> sink,
             final Function1<Throwable, Directive> supervisionStrategy,
             final ProducerTopicConfiguration topicConfiguration,
             final PolicyServiceAsyncProxy service) {
         // Get output topic from config
         Topic outputTopic = topicConfiguration.getTopics().get("output-topic");
+        Topic errorTopic = topicConfiguration.getTopics().get("error-topic");
 
         // Read messages from the stream source
+        //  return source
         return source
-                .map(message -> new Pair<>(message, Optional.ofNullable(message.record().value())))
+                .map(committableMessage -> new Pair<>(committableMessage, committableMessage.record().value()))
```

Comment:
I think this still needs to be an Optional because of the `START` and `END` messages as these will have a request value of `null`.

Reply:
The getResourceRules has been modified to handle a null.  All of the messages goes in there and, it creates a "container" object that reflects the state things with the PolicyRequest, Rules, Exception and the modified response.  Depending on the circumstances, some of these will be empty.  The original approach, some of this was being handled externally in the RunnableGraph and the the Optional was being used as part of the logic for if and when the getResourceRules was going to be called.